doc/nrf/ug_nrf9160.rst

```diff
@@ -110,9 +110,18 @@ See the `Mobile network operator certifications`_ for more information.
 There are two ways to update the modem firmware:
 
 Full upgrade
-  A full upgrade of the modem firmware requires a wired connection.
-  The upgrade is done through the nRF Connect Programmer, which is part of `nRF Connect for Desktop`_.
-  See `Updating the nRF9160 DK cellular modem`_ in the nRF Connect Programmer User Guide for instructions.
+  Both a wired or a wireless connection are suitable to proceed to a full upgrade of the modem firmware:
+
+  * When using a wired connection, you can use either the *nRF Connect Programmer*, as part of `nRF Connect for Desktop`_, or a custom application.
+    Both methods leverage the Simple Management Protocol (SMP) to provide an interface over UART which enables the device to perform the update.
+
+    The nRF Connect Programmer can be used to perform the update, regardless of the existing firmware.
```

Comment:
The way I interpreted this is that when not using the nRF connect programmer, the discriminating factors that would allow (or not) a user to perform the update would be the kind of images that are part of the existing firmware.

If one of the images contains a custom application similar to the sample linked a bit later in the text, then the update would be possible without the nrf connect programmer. If that would not be the case, only through the nrf connect programmer the user would be able to update the firmware.

@hakonfam is this correct?

Reply:
To avoid confusion I will try to state some simple facts here first, then I will answer the questions.

- nRF connect programmer can update the modem firmware regardless of the firmware currently programmed to the device. It loads some code to RAM and executes that AFAICR.
- The `pynrfjprog` python package provides an API for loading modem firmware to a device which is compatible with its "full modem DFU management" management protocol.
- The `lib_fmfu_mgmt` library implements the protocol mentioned above.
- The `fmfu_smp_svr_sample` shows the integration of the `lib_fmfu_mgmt` library.
- Alongside `fmfu_smp_svr_sample` there is a python script which shows the use of the `pynrfjprog`  python package.
- Hence, the python script mentioned above is able to update the modem firmware of a device _given_ that the `fmfu_smp_svr_sample` or any other sample which correctly integrates the `lib_fmfu_mgmt` library is programmed.

> You need some firmware, don't you?

Not when updating the modem firmware using nRF Connect programmer.

> The way I interpreted this is that when not using the nRF connect programmer, the discriminating factors that would allow (or not) a user to perform the update would be the kind of images that are part of the existing firmware.

If by "kind of images" you mean that the image must integrate the `lib_fmfu_mgmt` library, then yes.

> If one of the images contains a custom application similar to the sample linked a bit later in the text, then the update would be possible without the nrf connect programmer. If that's not the case, only through the nrf connect programmer the user would be able to update the firmware. 

Correct.

Reply:
ok :)

Reply:
Updated.